binr/rabin2/rabin2.c

```diff
@@ -898,12 +898,8 @@ int main(int argc, char **argv) {
 		if (!bits) {
 			bits = 32;
 		}
-		if (!r_bin_use_arch (bin, arch, bits, create)) {
-			eprintf ("Cannot set arch\n");
-			r_core_fini (&core);
-			return 1;
-		}
-		b = r_bin_create (bin, code, codelen, data, datalen);
+
+		b = r_bin_create (bin, code, codelen, data, datalen, arch, bits, create);
```

Comment:
we need a better solution to share the arch/cpu/bits/os information across all the modules from a single place. I dont like the idea of adding more parameters to that function, because we will need to set more options in the futre like OS, or any other parameters to create an executable, which are a lot.

Thats why it was using a fake rbinobj. this is similar approach from what i started with the RBinOptions and the open/close api model. we have to configure the bin plugin or pass an RBinOptions as argument with all that metadata/hints to create the file.

im also worried about the code+codelen,data+datalen parameters which may be probably better to use rbuffer and allow to pass multiple sections and different paddings, etc. so this should change also

Reply:
Yeah I was thinking the same about arch/bits/os/pluginname.... Do you have a suggested name for this new structure? I cannot think of any at the moment.

Reply:
wtf ror some reason my  answer didnt reached github wtf

i proposed RBinArchOptions or as long as it needs to be reused across all modules it should live in util and be named like RArchOptions

Reply:
It did reach github, I saw it. Anyway, why should it live in r_util? I didn't see any usage of arch/bits/etc. outside of RBin(or things that depend on RBin). I think arch/bits/etc. are very bin-specific concepts, so it makes sense to define them there.